Add Pinyin Support for Powertoys Run.

frg2089 opened the pull request:
<!-- Enter a brief description/summary of your PR here. What does it fix/what does it change/how was it tested (even manually, if necessary)? -->
## Summary of the Pull Request

Add Pinyin Support for Powertoys Run.

<!-- Please review the items on the PR checklist before submitting-->
## PR Checklist

- [x] **Closes:** #3587 #20370
- [ ] **Communication:** I've discussed this with core contributors already. If work hasn't been agreed, this work might be rejected
- [ ] **Tests:** Added/updated and all pass
- [ ] **Localization:** All end user facing strings can be localized
- [ ] **Dev docs:** Added/updated
- [ ] **New binaries:** Added on the required places
   - [ ] [JSON for signing](https://github.com/microsoft/PowerToys/blob/main/.pipelines/ESRPSigning_core.json) for new binaries
   - [x] [WXS for installer](https://github.com/microsoft/PowerToys/blob/main/installer/PowerToysSetup/Product.wxs) for new binaries and localization folder
   - [ ] [YML for CI pipeline](https://github.com/microsoft/PowerToys/blob/main/.pipelines/ci/templates/build-powertoys-steps.yml) for new test projects
   - [ ] [YML for signed pipeline](https://github.com/microsoft/PowerToys/blob/main/.pipelines/release.yml)
- [ ] **Documentation updated:** If checked, please file a pull request on [our docs repo](https://github.com/MicrosoftDocs/windows-uwp/tree/docs/hub/powertoys) and link it here: #xxx

<!-- Provide a more detailed description of the PR, other things fixed or any additional comments/features here -->
## Detailed Description of the Pull Request / Additional comments

Now, we can use Pinyin to search applications.

For example: `网易云音乐`(Netease Cloud Music)
Now you can search by `wyyyy` and `wangyiyunyinyue`

<!-- Describe how you validated the behavior. Add automated tests wherever possible, but list manual validation steps taken as well -->
## Validation Steps Performed

open the `PowerToys Run` and type into some pinyin.

frg2089 commented:
@microsoft-github-policy-service agree

crutkas commented on src/modules/launcher/Wox.Infrastructure/Wox.Infrastructure.csproj:
```diff
@@ -46,5 +46,6 @@
     <PackageReference Include="System.Drawing.Common" Version="6.0.0" />
     <PackageReference Include="System.IO.Abstractions" Version="12.2.5" />
     <PackageReference Include="System.Runtime" Version="4.3.1" />
+    <PackageReference Include="TinyPinyin.Net" Version="1.0.2" />
```
this is interesting, there are two variants here.  https://www.nuget.org/packages/TinyPinyin/ and https://www.nuget.org/packages/TinyPinyin.Net.  The version linked here does have a far higher usage count

htcfreek commented:
How does this work and what about string comparisons that happen inside the plugins?

frg2089 commented:
> How does this work and what about string comparisons that happen inside the plugins?

I recovered this PR #7455. And used another pinyin library.

frg2089 commented:
Hi all, I'm trying to modify the `settings.json` file to set the `ShouldUsePinyin` property to `true`, but it didn't work.

What should I do?

frg2089 commented:
It look like this.

![image](https://user-images.githubusercontent.com/42184238/199146148-7891c916-72fc-432c-b1eb-a401c3db7f1b.png)
![image](https://user-images.githubusercontent.com/42184238/199146153-916c97d3-00ad-45ba-99fc-b9bb461f6dce.png)
![image](https://user-images.githubusercontent.com/42184238/199146160-fe1d6bf9-a974-4518-828d-4ca14ec6e803.png)

for #20370
![image](https://user-images.githubusercontent.com/42184238/199146844-b41d595b-e303-4906-8c94-ab48af402597.png)

frg2089 committed:
Add Pinyin Support for Powertoys Run

using hyjiacan.pinyin4net

Revert microsoft/PowerToys#7455

closed microsoft/PowerToys#3587
closed microsoft/PowerToys#20370

Signed-off-by: 舰队的偶像-岛风酱! <frg2089@outlook.com>

frg2089 commented:
I'm using [hyjiacan.pinyin4net](https://www.nuget.org/packages/hyjiacan.pinyin4net) to support pinyin.

> 注：nuget 上还有一个 Pinyin4Net 的包， 那是有网友基于此项目的早期版本发布的，并非本人发布。
> Note: There is also a Pinyin4Net package on nuget, which was released by a user based on an earlier version of this project, not by me.

[hyjiacan.pinyin4net - ReadMe](https://github.com/hyjiacan/Pinyin4Net#nuget-%E5%AE%89%E8%A3%85)

jaimecbernardo committed:
Merge branch 'main' into main

jaimecbernardo committed:
Remove JetBrains annotations

jaimecbernardo reviewed (approved):
Thanks a lot for the contribution! This was long requested without anyone in the core team being speakers who could evaluate it.

I haven't tested the new functionality very specifically, but it looks like it doesn't break PTRun's normal functionality. Also couldn't notice any performance hit.
I've checked "hyjiacan.pinyin4net" source code and it looks benign.

Let's go with it and after we'll put it under a setting if we detect it's causing any issue.
